basesandbox/src/execution/executor.rs

```diff
@@ -0,0 +1,107 @@
+// Copyright 2020 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+
+use crate::ipc::semaphore::{Mode, Semaphore};
+use crate::ipc::Ipc;
+use std::process::Command;
+
+const SOCKET_SRC: &str = "./tmp/Foundry_Socket_Boxer";
+const SOCKET_DST: &str = "./tmp/Foundry_Socket_Boxee";
+const SEMAPHORE: &str = "/Foundry_Semaphore";
+
+struct Executor {
+    child: std::process::Child,
+}
+
+impl Executor {
+    fn new(path: &str, args: &[&str]) -> Self {
+        let mut command = Command::new(path);
+        command.args(args);
+        Executor {
+            child: command.spawn().unwrap(),
+        }
+    }
+}
+
+impl Drop for Executor {
+    fn drop(&mut self) {
+        for _ in 0..10 {
+            std::thread::sleep(std::time::Duration::from_millis(50));
```

Comment:
If you cannot reason why the child should finish in 500 milliseconds, you should leave the comment.